source/postprocess/particles.cc

```diff
@@ -461,11 +458,11 @@ namespace aspect
                       background_thread
                         = std::thread([&]()
                       {
-                        writer (filename, temporary_output_location, file_contents);
+                        writer (filename, temporary_output_location, *file_contents);
```

Comment:
This will not work. You start a new thread, and that thread has a reference to the `std::unique_ptr` object `file_contents`. But by the time the background thread dereferences it, (i) the string the pointer points to may already have been de-allocated, and (ii) the `std::unique_ptr` itself may no longer be a valid object. You have to capture everything you need here by copy, and you need to take over ownership of the pointer. In visualization.cc, this looks as follows:
```
                    = std::thread([ my_filename = std::move(filename),
                                    my_temporary_output_location = temporary_output_location,
                                    my_file_contents = std::move(file_contents)]()
                  {
                    writer (my_filename, my_temporary_output_location, *my_file_contents);
                  });
```

Reply:
I will note that this was *already* wrong for the `filename` and `temporary_output_location` variables. I think we only got lucky by the fact that the thread immediately calls `writer`, which *copies* these objects because these arguments are passed by value. This copy must have happened quickly enough that the main function did not already exit.

Reply:
hmm, oke. I did see that it was done slightly different in visualization.cc, but I wanted to keep the code as little as possible while introducing the unique_ptr. I will now just make it the same as there (which I probably should just have done from the start)